src/coreneuron/CMakeLists.txt

```diff
@@ -669,17 +669,19 @@ file(
   RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
   *.h *.hpp)
 
-configure_file("${CMAKE_BINARY_DIR}/generated/coreneuron/config/neuron_version.hpp"
-               "${CMAKE_BINARY_DIR}/include/coreneuron/config/neuron_version.hpp" COPYONLY)
+cpp_cc_build_time_copy(INPUT ${CMAKE_BINARY_DIR}/generated/coreneuron/config/neuron_version.hpp
+                       OUTPUT ${CMAKE_BINARY_DIR}/include/coreneuron/config/neuron_version.hpp)
 foreach(header ${main_headers})
-  configure_file("${header}" "${CMAKE_BINARY_DIR}/include/coreneuron/${header}" COPYONLY)
+  cpp_cc_build_time_copy(INPUT ${CMAKE_CURRENT_SOURCE_DIR}/${header}
+                         OUTPUT ${CMAKE_BINARY_DIR}/include/coreneuron/${header})
 endforeach()
```

Comment:
@1uc:  I think we have a problem/bug here: IIUC, `cpp_cc_build_time_copy` is copying the files during the build and new targets are created internally. We are not setting any dependencies.

But, these headers need to be available when `nrnivmodl-core` target is executed i.e. ` nrnivmodl-core` needs to be executed after these ' cpp_cc_build_time_copy` are executed.

Because of this, I believe we are seeing errors that @heerener saw on graviton:

```console
....
[NMODL] [info] :: Running cnexp visitor
[NMODL] [info] :: Running C backend code generator
aarch64/corenrn/mod2c/exp2syn.cpp:17:10: fatal error: coreneuron/nrnconf.h: No such file or directory
   17 | #include <coreneuron/nrnconf.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [/tmp/root/spack-stage/spack-stage-neuron-9.0.a11-g6jmgi2bu34wne42qvtrsr2ja7rqcse7/spack-build-g6jmgi2/share/coreneuron/nrnivmodl_core_makefile:184: aarch64/corenrn/build/exp2syn.o] Error 1
make: *** Waiting for unfinished jobs....
aarch64/corenrn/mod2c/expsyn.cpp:17:10: fatal error: coreneuron/nrnconf.h: No such file or directory
   17 | #include <coreneuron/nrnconf.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [/tmp/root/spack-stage/spack-stage-neuron-9.0.a11-g6jmgi2bu34wne42qvtrsr2ja7rqcse7/spack-build-g6jmgi2/share/coreneuron/nrnivmodl_core_makefile:184: aarch64/corenrn/build/expsyn.o] Error 1
aarch64/corenrn/mod2c/hh.cpp:17:10: fatal error: coreneuron/nrnconf.h: No such file or directory
   17 | #include <coreneuron/nrnconf.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
That's unfortunate, I missed that aspect. The places this PR changes `configure_file` to `cpp_cc_build_time_copy` I think are safe to revert. I wanted to avoid rerunning the configure stage. However, any changes to those files should trigger a reconfigure. The places were we replace `file(COPY` I'd keep or change to `configure_file`, because those only happen once will lead to partial builds.

Reply:
Sounds good to me! (before reverting, I wanted to see if you see any other alternatives. But I agree that we don’t need complex solution)